cloudpickle/cloudpickle.py

```diff
@@ -141,8 +143,22 @@ def _whichmodule(obj, name):
     return None
 
 
+if sys.version_info[:2] < (3, 7):
+    def _get_module_attr(obj):
+        if isinstance(obj, typing.TypeVar):
+            return None
+        return getattr(obj, '__module__', None)
+else:
+    def _get_module_attr(obj):
+        return getattr(obj, '__module__', None)
+
+
 def _is_global(obj, name=None):
     """Determine if obj can be pickled as attribute of a file-backed module"""
+    return _get_module_if_global(obj, name)
+
+
+def _get_module_if_global(obj, name=None):
     if name is None:
         name = getattr(obj, '__qualname__', None)
     if name is None:
```

Comment:
It would be nice to insert a comment to give examples on when `getattr(obj, '__name__', None)` is expected to be not None while `getattr(obj, '__qualname__', None) is None` does not exists in recent versions of Python.

Reply:
FYI, I tried to run the test with the new following new line:

```diff
diff --git a/cloudpickle/cloudpickle.py b/cloudpickle/cloudpickle.py
index 996a002..1dceca6 100644
--- a/cloudpickle/cloudpickle.py
+++ b/cloudpickle/cloudpickle.py
@@ -163,6 +163,8 @@ def _get_module_if_global(obj, name=None):
         name = getattr(obj, '__qualname__', None)
     if name is None:
         name = getattr(obj, '__name__', None)
+        raise ValueError("%s (%s) had no __qualname__ but __name__ is %s"
+                         % (obj, type(obj), name))
 
     module_name = _whichmodule(obj, name)
 
```

and the only tests that fail are the TypeVar tests. So indeed this case is need for TypeVar objects that have a `__name__` attribute instead of a `__qualname__` attribute.

Reply:
Good catch. Previously, `__name__` was there for Py2 support, I think.

Reply:
I believe so as well.